tools/experiments/experiments.js

```diff
@@ -237,24 +268,31 @@ function updateExperimentRow(experiment) {
  * @return {boolean}
  */
 function isExperimentOn_(id) {
-  if (id == CANARY_EXPERIMENT_ID) {
-    return getCookie(window, 'AMP_CANARY') == AMP_CANARY_COOKIE.CANARY;
-  } else if (id == RC_EXPERIMENT_ID) {
-    return getCookie(window, 'AMP_CANARY') == AMP_CANARY_COOKIE.RC;
+  switch (id) {
+    case EXPERIMENTAL_CHANNEL_ID:
+      return getCookie(window, 'AMP_CANARY') == AMP_CANARY_COOKIE.CANARY;
+    case BETA_CHANNEL_ID:
+      return getCookie(window, 'AMP_CANARY') == AMP_CANARY_COOKIE.RC;
+    case RTV_CHANNEL_ID:
+      return RTV_PATTERN.test(getCookie(window, 'AMP_CANARY'));
+    default:
+      return isExperimentOn(window, /*OK*/ id);
   }
-  return isExperimentOn(window, /*OK*/ id);
 }
 
 /**
  * Opts in to / out of the "canary" or "rc" runtime types by setting the
  * AMP_CANARY cookie.
- * @param {string} cookieState One of AMP_CANARY_COOKIE.{DISABLED|CANARY|RC}
+ * @param {string} cookieState One of AMP_CANARY_COOKIE.{DISABLED|CANARY|RC} or
+ *   a 15-digit RTV.
  */
 function setAmpCanaryCookie_(cookieState) {
-  const validUntil =
-    cookieState != AMP_CANARY_COOKIE.DISABLED
-      ? Date.now() + COOKIE_MAX_AGE_MS
-      : 0;
+  let validUntil = 0;
+  if (RTV_PATTERN.test(cookieState)) {
+    validUntil = Date.now() + RTV_COOKIE_MAX_AGE_MS;
+  } else if (cookieState != AMP_CANARY_COOKIE.DISABLED) {
+    validUntil = Date.now() + COOKIE_MAX_AGE_MS;
```

Comment:
If someone enters an RTV that doesn't match the correct 13 digit format, it will fall through here. Should there be a separate error handling case before this `else` block?

Reply:
This method gets called post-validation, at this point it only receives one of the hard-coded states, or a 15-digit RTV